src/client.ts

```diff
@@ -0,0 +1,132 @@
+import * as request from "request-promise-native";
+
+interface assoc {
+    [key: string]: any;
+}
+
+export class Client {
+
+    endpoint: string = 'https://appwrite.io/v1';
+    headers: assoc = {
+        'content-type': '',
+        'x-sdk-version': 'appwrite:typescript:',
+    };
+    
+    /**
+     * Set Project
+     *
+     * Your project ID
+     *
+     * @param string value
+     *
+     * @return self
+     */
+    setProject(value: string): this {
+        this.addHeader('X-Appwrite-Project', value);
+
+        return this;
+    }
+
+    /**
+     * Set Key
+     *
+     * Your secret API key
+     *
+     * @param string value
+     *
+     * @return self
+     */
+    setKey(value: string): this {
+        this.addHeader('X-Appwrite-Key', value);
+
+        return this;
+    }
+
+    /**
+     * Set Locale
+     *
+     * @param string value
+     *
+     * @return self
+     */
+    setLocale(value: string): this {
+        this.addHeader('X-Appwrite-Locale', value);
+
+        return this;
+    }
+
+    /**
+     * Set Mode
+     *
+     * @param string value
+     *
+     * @return self
+     */
+    setMode(value: string): this {
+        this.addHeader('X-Appwrite-Mode', value);
+
+        return this;
+    }
+
+
+    /***
+     * @param endpoint
+     * @return this
+     */
+    setEndpoint(endpoint: string): this {
+        this.endpoint = endpoint;
+
+        return this;
+    }
+
+    /**
+     * @param key string
+     * @param value string
+     */
+    addHeader(key: string, value: string): this {
+        this.headers[key.toLowerCase()] = value.toLowerCase();
+        
+        return this;
+    }
+
+    async call(method: string, path: string = '', headers: assoc = {}, params: assoc = {}) {
```

Comment:
It'd be very confusing to call it this way. Consider replacing `headers` and `params` in a single `options` argument which is an object with `headers` and `params` properties.

Reply:
Also, you can better type `method` as `method: 'GET' | 'POST' | 'PUT' | 'DELETE' | 'PATCH'` (string literal). This makes sure that the user passes only one of these values, so no need for the `toUpperCase()` below (though it's not needed anyway IIRC), and would also ensure that passing an invalid HTTP verb would fail at compile-time, not runtime.